src/Sarif.Driver/Sdk/AnalyzeOptionsBase.cs

```diff
@@ -113,36 +111,18 @@
         [Option(
             "level",
             Separator = ';',
-            Default = new FailureLevel[] { FailureLevel.Error, FailureLevel.Warning },
             HelpText = "A semicolon delimited list to filter output of scan results to one or more failure levels. Valid values: Error, Warning and Note.")]
         public IEnumerable<FailureLevel> Level { get; set; }
 
-        private FailureLevelSet failureLevels;
```

Comment:
>FailureLevelSet [](http://example.com/codeflow?start=15&length=15)

Providing defaults in the options classes is problematic because it interferes with the context object default value functionality (which is much richer/close to the analysis model.). In the options classes, we should actually return null for all cases where the property isn't explicit on the command-line, otherwise reflect the command-line precisely. That's the model. There's more clean-up to do.